### PR TITLE
Added json to report from Qeb-hwt

### PR DESCRIPTION
**Pull request by saisankargochhayat**
## Related Issues and Dependencies
Related to - https://github.com/thoth-station/qeb-hwt/issues/63

### CODE REVIEW

File: thoth/report_processing/pretty_report.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    if "requires" in requirements:
+        if requirements["requires"]:
+            md += "\n\n" + "Requires in Advised Pipfile"
+            df = pd.DataFrame([requirements["requires"]])
+            md += "\n\n" + df.to_markdown()
+
+    if "source" in requirements:
+        if requirements["source"]:
+            md += "\n\n" + "Source in Advised Pipfile"
+            df = pd.DataFrame(requirements["source"])
```
Comment (bissenbay) [inline thoth/report_processing/pretty_report.py]:
```suggestion
            df = pd.DataFrame([requirements["source"]])
```

File: thoth/report_processing/pretty_report.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        md_report += "\n\n" + "Packages in Advised Pipfile"
+
+    packages_names = []
+    packages_versions = []
+
+    for package_name, requested_version in product["project"]["requirements"][packages].items():
+        packages_names.append(package_name)
+        packages_versions.append(requested_version)
+
+    data = {"package_name": packages_names, "pakage_version": packages_versions}
```
Comment (bissenbay) [inline thoth/report_processing/pretty_report.py]:
```suggestion
    data = {"package_name": packages_names, "package_version": packages_versions}
```

**Review by bissenbay — COMMENTED** [review]
LGTM

File: thoth/report_processing/pretty_report.py (hunk 2 above)
Comment (saisankargochhayat) [reply thoth/report_processing/pretty_report.py]: Thank you nice catch!

File: thoth/report_processing/pretty_report.py (hunk 1 above)
Comment (saisankargochhayat) [reply thoth/report_processing/pretty_report.py]: I think the value of requirements["source"] is an arrary. For example below .. 
line 124 in https://gist.github.com/saisankargochhayat/32fc67949427d200ce5fabe4e15036df

File: thoth/report_processing/pretty_report.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Utils Create Markdown output from adviser report input."""
+
+import pandas as pd
+from typing import Dict, Any
+
+
```
Comment (pacospace) [inline thoth/report_processing/pretty_report.py]: What about calling the file `adviser.py` where we have a `class Adviser` that contains all methods we can use on adviser report? :)

Comment (saisankargochhayat) [reply thoth/report_processing/pretty_report.py]: That sounds reasonable, I will change it accordingly.

File: thoth/report_processing/adviser.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                    final_df.loc[counter] = pd.DataFrame([justification]).iloc[0]
+                    counter += 1
+
+        md += "\n\n" + final_df.to_markdown()
+
+        if is_justification:
+            return md
+
+        # Packages in Advised Pipfile
+        md += Adviser._add_packages_in_advised_pipfile_to_md_report(product=product, is_dev=False)
```
Comment (pacospace) [inline thoth/report_processing/adviser.py:63]: you can use `self.` instead of `Adviser.` for methods under the class :)

**Review by pacospace — COMMENTED** [review]
One small comment ;)

File: thoth/report_processing/adviser.py (hunk 4 above)
Comment (saisankargochhayat) [reply thoth/report_processing/adviser.py:63]: I don't think static methods can be accessed using self, do you want me to change them to `classmethods`?

Comment (pacospace) [reply thoth/report_processing/adviser.py:63]: right, didn't see that sorry :( no it is very good then :) Thanks!!

**Review by pacospace — APPROVED** [review]

File: thoth/report_processing/adviser.py (hunk 4 above)
Comment (fridex) [reply thoth/report_processing/adviser.py:63]: All of these will work:

```python
class A:

    @staticmethod
    def s():
        pass

    @classmethod
    def c(cls):
        # Can call cls.s() or cls.c()
        pass

    def m(self):
        # Can call self.s() or self.c() or self.m()
        pass


A.s()
A.c()
obj = A()
obj.s()
obj.c()
obj.m()
```

Comment (saisankargochhayat) [reply thoth/report_processing/adviser.py:63]: Thanks :small_airplane:

**Pull request merged**